### PR TITLE
[SPARK-26245][SQL] Add Float literal

### PR DESCRIPTION
**Pull request by wangyum**
## What changes were proposed in this pull request?

This PR adds parser support for `Float` literals.

Hive support this feature:
image

## How was this patch tested?

unit tests

### CODE REVIEW

**Comment by wangyum** [comment]
retest this please

**Comment by wangyum** [comment]
retest this please

File: sql/core/src/test/resources/sql-tests/inputs/literals.sql [hunk 1]
```diff
@@ -53,6 +53,10 @@ select .e3;
 -- very large decimals (overflowing double).
 select 1E309, -1E309;
 
+-- float
+select 0F, 1F, 1.2F, -1F, -1.2F, 3.4028235E30F;
+select 3.4028235E39F;
```
Comment (gatorsmile) [inline sql/core/src/test/resources/sql-tests/inputs/literals.sql]: Could you move the test cases to the end of the file?

Also clearly specify the positive and negative test cases in the comments?

More test cases are needed.

File: sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4 [hunk 2]
```diff
@@ -1045,6 +1046,11 @@ DOUBLE_LITERAL
     | DECIMAL_DIGITS EXPONENT? 'D' {isValidDecimal()}?
     ;
 
+FLOAT_LITERAL
+    : DIGIT+ EXPONENT? 'F'
+    | DECIMAL_DIGITS EXPONENT? 'F' {isValidDecimal()}?
```
Comment (gatorsmile) [inline sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4]: Could you check who supports these Float literals? What are their restrictions? 

The ANSI SQL standard does not have such a literal.

Comment (maropu) [reply sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4]: This is implementation-specific;
```
hive> create temporary table t1 as select 1.0F;
hive> describe t1;
OK
f                   	decimal(1,0)  

hive> create temporary table t2 as select 1.0D;
hive> describe t2;
OK
_c0                 	double  
```
```
postgres=# create temporary table t1 as select 1.0F;
postgres=# \d t1
 f      | numeric | 

postgres=# create temporary table t2 as select 1.0D;
postgres=# \d t2
 d      | numeric | 

```
```
mysql> create temporary table t1 as select 1.0F;
mysql> describe t1;
+-------+--------------+------+-----+---------+-------+
| Field | Type         | Null | Key | Default | Extra |
+-------+--------------+------+-----+---------+-------+
| F     | decimal(2,1) | NO   |     | 0.0     | NULL  |
+-------+--------------+------+-----+---------+-------+

mysql> create temporary table t2 as select 1.0D;
mysql> describe t2;
+-------+--------------+------+-----+---------+-------+
| Field | Type         | Null | Key | Default | Extra |
+-------+--------------+------+-----+---------+-------+
| D     | decimal(2,1) | NO   |     | 0.0     | NULL  |
+-------+--------------+------+-----+---------+-------+
```

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
Hi, @wangyum .
In Hive 2.3, `1.0D` is considered as Double, but `1.0F` is considered as `1.0 F` where `F` is an alias.
In Hive 1.2, `1.0D` and `1.0F` are considered as `1.0 D` and `1.0 F` where `D` and `F` are aliases.
So, this is not for Hive compatibility. We had better close this PR.

**Pull request closed without merging**